### DISCUSSION TITLE
T&H Action Items by IC

### DISCUSSION THREAD

**Discussion by will-corcoran** [stacks-network/stacks-core#3800] [Quality of Life]
👋 Hi All - 

At the conclusion of Working Group calls I will update your discussion item below (organized by IC) with any action items brought up in the call.  This will be the catch-all for these items until they can make their way to the [Project Board](https://github.com/orgs/stacks-network/projects/63). 

Any time you are tagged here - if possible - please reply and link me to the related Issue / PR so I can educate myself further and get it added to the board.

Thanks!

**Comment by will-corcoran** [comment]
@kantai here are your action items.  see note at top of thread and comment accordingly.  

- Complete Skiff document for QoL and T&H.  let @will-at-stacks know when done so I can summarize on slides for Monday.
- Comment here to provide me with any suggested questions for the survey we discussed.
- Identify any starter tasks for T&H QoL ICs whenever possible.

**Reply by kantai** [reply]
## Suggestions for Code Quality Survey

* How likely are you to recommend contributing to the stacks-blockchain to another developer?
* If you have been interested in contributing to the stacks-blockchain but were unable to or dissuaded from doing so, what did you want to contribute, how much progress did you make, and what prevented you from contributing?
* In the past two weeks, how often has your progress on a task been delayed by a surprising issue in the stacks-blockchain codebase?
* What is one task or process while interacting with the blockchain that you found more frustrating than you expected? Try to be as specific as possible. 
* If you could change one thing about the implementation of the stacks-blockchain or its repository, what would it be?

## Starter Tasks for Testing and Hardening

* Property testing for PoX-2 and PoX-3 using clarinet built with pox locking support (see discussion https://github.com/stacks-network/stacks-blockchain/discussions/3732 and the PR for clarinet with locking support https://github.com/hirosystems/clarinet/pull/1074)
* Sufficient automatic epoch-gated testing that the trait bug necessitating Stacks 2.3 would have been caught (as discussed in https://github.com/stacks-network/stacks-blockchain/discussions/3732)

**Comment by will-corcoran** [comment]
@sergey-shandar here are your action items. see note at top of thread and comment accordingly.

- Create a discussion thread or documentation to help encourage and guide others to write instructions for tests without automation, helping identify what can be automated or improved upon.
- Work with @moodmosaic to get him onboarded first ASAP so he can start helping with Clarity VM testing
- Provide @bitfari and @CAGS295 with an update as to when you would like them to start onboarding.
- Nikos, Carlos, and Jordhy: please comment here and let us know your availability and when you can start your 3 month appointment.

**Reply by will-corcoran** [reply]
@obycode pinging you here as the item above touches Clarity VM

**Reply by CAGS295** [reply]
I can start asap. My timezone is UTC-6.

**Reply by moodmosaic** [reply]
Thank you for the opportunity. I am available to start work on the project immediately, with the exception of the upcoming week due to prior commitments.

I would also like to note that I am based in Europe, operating primarily in the Central European Time (CET) zone. Please consider this when scheduling any meetings. Despite any potential time difference, I remain flexible and am willing to make necessary adjustments to accommodate the project's needs.

**Comment by will-corcoran** [comment]
@ashtonstephens here are your action items. see note at top of thread and comment accordingly.

- Working on [ticket 466](https://github.com/Trust-Machines/stacks-sbtc/issues/466) related to ensuring Mac OS latest tests work properly and document any issues encountered.

**Comment by will-corcoran** [comment]
@jo-tm here are your action items. see note at top of thread and comment accordingly.

- Investigate and fix the code coverage module issue for Github clarity information request.
- Test the clarity VM box and understand if any bugs are related to clarinet or live blockchain testing.

**Reply by will-corcoran** [reply]
@obycode pinging you here as the item above touches Clarity VM

**Comment by will-corcoran** [comment]
@wileyj here are your action items. see note at top of thread and comment accordingly.

- Working on PR for the CI changes (still working on some of the slower tests) (working on targeting).  

(please link here for me)

**Comment by will-corcoran** [comment]
@CharlieC3 here are your action items. see note at top of thread and comment accordingly.

- Assist @wileyj  on CI or Hiro Infra (pre-2.4 release issue) NFII (Terraform CDK).

**Comment by will-corcoran** [comment]
@jbencin here are your action items. see note at top of thread and comment accordingly.

- Review and explore implementing automated end-to-end testing using clarinet integrate repository's pox tests.

**Comment by will-corcoran** [comment]
@fpbgg here are your action items. see note at top of thread and comment accordingly.

- Leads mutation tests.  Please provide an update on any current efforts.

**Comment by will-corcoran** [comment]
@deantchi here are your action items. see note at top of thread and comment accordingly.

- As noted on the call you play a similar role as @CharlieC3  (gets more involved as we get closer to release - works with @wileyj , covers unique tests).

**Comment by will-corcoran** [comment]
@kenrogers here are your action items. see note at top of thread and comment accordingly.

- Taking the Notion docs that @saralab provided and converting the curriculum into a cohesive GitHub page or repo or Wiki.
- Exploring options for progress tracking so new hires / contributors can track their progress easily.  you might chat @setzeus and @setbern - they might be working on something similar for Bitscript...

**Comment by will-corcoran** [comment]
👀  @muneeb-ali @netrome @cuevasm @andreserrano @jcnelson  see this thread.  tracking action items from Working Group calls in a thread like this for each WG.  This will serve as a catch-all until I can capture on the [Project Board](https://github.com/orgs/stacks-network/projects/63/views/6).
